File: java/src/main/java/com/openalpr/api/models/InlineResponse200.java
package com.openalpr.api.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import com.openalpr.api.models.AlprPlate;
import com.openalpr.api.models.Classification;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.math.BigDecimal;
import java.util.*;





@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-02-17T14:18:32.383-05:00")
public class InlineResponse200   {
  
  private Integer creditsMonthlyUsed = null;
  private Integer creditsMonthlyTotal = null;
  private Integer imgWidth = null;
  private List<Classification> color = new ArrayList<Classification>();
  private List<Classification> makemodel = new ArrayList<Classification>();
  private BigDecimal totalProcessingTime = null;
  private Integer imgHeight = null;
  private Integer creditCost = null;
  private AlprPlate plate = null;
  private List<Classification> make = new ArrayList<Classification>();

  
  /**
   * The number of API credits used this month
   **/
  
  @ApiModelProperty(value = "The number of API credits used this month")
  @JsonProperty("credits_monthly_used")
  public Integer getCreditsMonthlyUsed() {
    return creditsMonthlyUsed;
  }
  public void setCreditsMonthlyUsed(Integer creditsMonthlyUsed) {
    this.creditsMonthlyUsed = creditsMonthlyUsed;
  }

  
  /**
   * The maximum number of API credits available this month according to your plan
   **/
  
  @ApiModelProperty(value = "The maximum number of API credits available this month according to your plan")
  @JsonProperty("credits_monthly_total")
  public Integer getCreditsMonthlyTotal() {
    return creditsMonthlyTotal;
  }
  public void setCreditsMonthlyTotal(Integer creditsMonthlyTotal) {
    this.creditsMonthlyTotal = creditsMonthlyTotal;
  }

  
  /**
   * Width of the uploaded image in pixels
   **/
  
  @ApiModelProperty(value = "Width of the uploaded image in pixels")
  @JsonProperty("img_width")
  public Integer getImgWidth() {
    return imgWidth;
  }
  public void setImgWidth(Integer imgWidth) {
    this.imgWidth = imgWidth;
  }

  
  /**
   * Results from color vehicle analysis task
   **/
  
  @ApiModelProperty(value = "Results from color vehicle analysis task")
  @JsonProperty("color")
  public List<Classification> getColor() {
    return color;
  }
  public void setColor(List<Classification> color) {
    this.color = color;
  }

  
  /**
   * Results from makemodel vehicle analysis task
   **/
  
  @ApiModelProperty(value = "Results from makemodel vehicle analysis task")
  @JsonProperty("makemodel")
  public List<Classification> getMakemodel() {
    return makemodel;
  }
  public void setMakemodel(List<Classification> makemodel) {
    this.makemodel = makemodel;
  }

  
  /**
   * Time spent processing all tasks (in milliseconds)
   **/
  
  @ApiModelProperty(value = "Time spent processing all tasks (in milliseconds)")
  @JsonProperty("total_processing_time")
  public BigDecimal getTotalProcessingTime() {
    return totalProcessingTime;
  }
  public void setTotalProcessingTime(BigDecimal totalProcessingTime) {
    this.totalProcessingTime = totalProcessingTime;
  }

  
  /**
   * Height of the input image in pixels
   **/
  
  @ApiModelProperty(value = "Height of the input image in pixels")
  @JsonProperty("img_height")
  public Integer getImgHeight() {
    return imgHeight;
  }
  public void setImgHeight(Integer imgHeight) {
    this.imgHeight = imgHeight;
  }

  
  /**
   * The number of API credits that were used to process this image
   **/
  
  @ApiModelProperty(value = "The number of API credits that were used to process this image")
  @JsonProperty("credit_cost")
  public Integer getCreditCost() {
    return creditCost;
  }
  public void setCreditCost(Integer creditCost) {
    this.creditCost = creditCost;
  }

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("plate")
  public AlprPlate getPlate() {
    return plate;
  }
  public void setPlate(AlprPlate plate) {
    this.plate = plate;
  }

  
  /**
   * Results from make vehicle analysis task
   **/
  
  @ApiModelProperty(value = "Results from make vehicle analysis task")
  @JsonProperty("make")
  public List<Classification> getMake() {
    return make;
  }
  public void setMake(List<Classification> make) {
    this.make = make;
  }

  

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    InlineResponse200 inlineResponse200 = (InlineResponse200) o;
    return Objects.equals(creditsMonthlyUsed, inlineResponse200.creditsMonthlyUsed) &&
        Objects.equals(creditsMonthlyTotal, inlineResponse200.creditsMonthlyTotal) &&
        Objects.equals(imgWidth, inlineResponse200.imgWidth) &&
        Objects.equals(color, inlineResponse200.color) &&
        Objects.equals(makemodel, inlineResponse200.makemodel) &&
        Objects.equals(totalProcessingTime, inlineResponse200.totalProcessingTime) &&
        Objects.equals(imgHeight, inlineResponse200.imgHeight) &&
        Objects.equals(creditCost, inlineResponse200.creditCost) &&
        Objects.equals(plate, inlineResponse200.plate) &&
        Objects.equals(make, inlineResponse200.make);
  }

  @Override
  public int hashCode() {
    return Objects.hash(creditsMonthlyUsed, creditsMonthlyTotal, imgWidth, color, makemodel, totalProcessingTime, imgHeight, creditCost, plate, make);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class InlineResponse200 {\n");
    
    sb.append("    creditsMonthlyUsed: ").append(toIndentedString(creditsMonthlyUsed)).append("\n");
    sb.append("    creditsMonthlyTotal: ").append(toIndentedString(creditsMonthlyTotal)).append("\n");
    sb.append("    imgWidth: ").append(toIndentedString(imgWidth)).append("\n");
    sb.append("    color: ").append(toIndentedString(color)).append("\n");
    sb.append("    makemodel: ").append(toIndentedString(makemodel)).append("\n");
    sb.append("    totalProcessingTime: ").append(toIndentedString(totalProcessingTime)).append("\n");
    sb.append("    imgHeight: ").append(toIndentedString(imgHeight)).append("\n");
    sb.append("    creditCost: ").append(toIndentedString(creditCost)).append("\n");
    sb.append("    plate: ").append(toIndentedString(plate)).append("\n");
    sb.append("    make: ").append(toIndentedString(make)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: javascript/src/model/InlineResponse200.js
// require files in Node.js environment
var AlprPlate;var Classification;
if (typeof module === 'object' && module.exports) {
  
  AlprPlate = require('./AlprPlate.js');
  Classification = require('./Classification.js');
}




//export module
if ( typeof define === "function" && define.amd ) {
  define('InlineResponse200', ['jquery', 'Array', 'Array', 'AlprPlate', 'Array'],
    function($, Array, Array, AlprPlateArray) {
      return InlineResponse200;
   });
}


var InlineResponse200 = function InlineResponse200() { 
  var self = this;
  
  /**
   * The number of API credits used this month
   * datatype: Integer
   **/
  self.creditsMonthlyUsed = null;
  
  /**
   * The maximum number of API credits available this month according to your plan
   * datatype: Integer
   **/
  self.creditsMonthlyTotal = null;
  
  /**
   * Width of the uploaded image in pixels
   * datatype: Integer
   **/
  self.imgWidth = null;
  
  /**
   * Results from color vehicle analysis task
   * datatype: Array
   **/
  self.color = [];
  
  /**
   * Results from makemodel vehicle analysis task
   * datatype: Array
   **/
  self.makemodel = [];
  
  /**
   * Time spent processing all tasks (in milliseconds)
   * datatype: Number
   **/
  self.totalProcessingTime = null;
  
  /**
   * Height of the input image in pixels
   * datatype: Integer
   **/
  self.imgHeight = null;
  
  /**
   * The number of API credits that were used to process this image
   * datatype: Integer
   **/
  self.creditCost = null;
  
  /**
   * datatype: AlprPlate
   **/
  self.plate = null;
  
  /**
   * Results from make vehicle analysis task
   * datatype: Array
   **/
  self.make = [];
  
  
  self.constructFromObject = function(data) {
    if (!data) {
      return;
    }
    
    self.creditsMonthlyUsed = data.credits_monthly_used;
    
    self.creditsMonthlyTotal = data.credits_monthly_total;
    
    self.imgWidth = data.img_width;
    
    self.color = data.color;
    
    self.makemodel = data.makemodel;
    
    self.totalProcessingTime = data.total_processing_time;
    
    self.imgHeight = data.img_height;
    
    self.creditCost = data.credit_cost;
    
    self.plate = data.plate;
    
    self.make = data.make;
    
  }

  
  /**
   * get The number of API credits used this month
   * @return {Integer}
   **/
  self.getCreditsMonthlyUsed = function() {
    return self.creditsMonthlyUsed;
  }

  /**
   * set The number of API credits used this month
   * @param {Integer} creditsMonthlyUsed
   **/
  self.setCreditsMonthlyUsed = function (creditsMonthlyUsed) {
    self.creditsMonthlyUsed = creditsMonthlyUsed;
  }
  
  /**
   * get The maximum number of API credits available this month according to your plan
   * @return {Integer}
   **/
  self.getCreditsMonthlyTotal = function() {
    return self.creditsMonthlyTotal;
  }

  /**
   * set The maximum number of API credits available this month according to your plan
   * @param {Integer} creditsMonthlyTotal
   **/
  self.setCreditsMonthlyTotal = function (creditsMonthlyTotal) {
    self.creditsMonthlyTotal = creditsMonthlyTotal;
  }
  
  /**
   * get Width of the uploaded image in pixels
   * @return {Integer}
   **/
  self.getImgWidth = function() {
    return self.imgWidth;
  }

  /**
   * set Width of the uploaded image in pixels
   * @param {Integer} imgWidth
   **/
  self.setImgWidth = function (imgWidth) {
    self.imgWidth = imgWidth;
  }
  
  /**
   * get Results from color vehicle analysis task
   * @return {Array}
   **/
  self.getColor = function() {
    return self.color;
  }

  /**
   * set Results from color vehicle analysis task
   * @param {Array} color
   **/
  self.setColor = function (color) {
    self.color = color;
  }
  
  /**
   * get Results from makemodel vehicle analysis task
   * @return {Array}
   **/
  self.getMakemodel = function() {
    return self.makemodel;
  }

  /**
   * set Results from makemodel vehicle analysis task
   * @param {Array} makemodel
   **/
  self.setMakemodel = function (makemodel) {
    self.makemodel = makemodel;
  }
  
  /**
   * get Time spent processing all tasks (in milliseconds)
   * @return {Number}
   **/
  self.getTotalProcessingTime = function() {
    return self.totalProcessingTime;
  }

  /**
   * set Time spent processing all tasks (in milliseconds)
   * @param {Number} totalProcessingTime
   **/
  self.setTotalProcessingTime = function (totalProcessingTime) {
    self.totalProcessingTime = totalProcessingTime;
  }
  
  /**
   * get Height of the input image in pixels
   * @return {Integer}
   **/
  self.getImgHeight = function() {
    return self.imgHeight;
  }

  /**
   * set Height of the input image in pixels
   * @param {Integer} imgHeight
   **/
  self.setImgHeight = function (imgHeight) {
    self.imgHeight = imgHeight;
  }
  
  /**
   * get The number of API credits that were used to process this image
   * @return {Integer}
   **/
  self.getCreditCost = function() {
    return self.creditCost;
  }

  /**
   * set The number of API credits that were used to process this image
   * @param {Integer} creditCost
   **/
  self.setCreditCost = function (creditCost) {
    self.creditCost = creditCost;
  }
  
  /**
   * @return {AlprPlate}
   **/
  self.getPlate = function() {
    return self.plate;
  }

  /**
   * @param {AlprPlate} plate
   **/
  self.setPlate = function (plate) {
    self.plate = plate;
  }
  
  /**
   * get Results from make vehicle analysis task
   * @return {Array}
   **/
  self.getMake = function() {
    return self.make;
  }

  /**
   * set Results from make vehicle analysis task
   * @param {Array} make
   **/
  self.setMake = function (make) {
    self.make = make;
  }
  

  self.toJson = function () {
    return JSON.stringify(self);
  }
}

if (typeof module === 'object' && module.exports) {
  module.exports = InlineResponse200;
}
